composition-js/src/merging/merge.ts

```diff
@@ -2954,6 +2981,182 @@ class Merger {
     });
   }
 
+  private recordTypeAppliedDirectivesToMerge(sources: Sources<SchemaElement<any, any>>, dest: SchemaElement<any, any>) {
+    const inaccessibleName = this.inaccessibleDirectiveInSupergraph?.name;
+    const names = this.gatherAppliedDirectiveNames(sources);
+
+    if (inaccessibleName && names.has(inaccessibleName)) {
+      this.mergeAppliedDirective(inaccessibleName, sources, dest);
+      names.delete(inaccessibleName);
+    }
+
+    if (dest instanceof InterfaceType) {
+      // we need to propagate access control from implementing types upward to the interface
+      for (const { name, nameInSupergraph } of this.accessControlDirectivesInSupergraph) {
+        if (nameInSupergraph) {
```

Comment:
If you changed `nameInSupergraph` to being non-`undefined` as mentioned above, you won't need this `if` check.